server/src/main/java/org/apache/druid/server/metrics/QueryCountStatsMonitor.java

```diff
@@ -21,24 +21,30 @@
 
 import com.google.common.collect.ImmutableMap;
 import com.google.inject.Inject;
+import org.apache.druid.collections.BlockingPool;
+import org.apache.druid.guice.annotations.Merging;
 import org.apache.druid.java.util.emitter.service.ServiceEmitter;
 import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
 import org.apache.druid.java.util.metrics.AbstractMonitor;
 import org.apache.druid.java.util.metrics.KeyedDiff;
 
+import java.nio.ByteBuffer;
 import java.util.Map;
 
 public class QueryCountStatsMonitor extends AbstractMonitor
 {
   private final KeyedDiff keyedDiff = new KeyedDiff();
   private final QueryCountStatsProvider statsProvider;
+  private final BlockingPool<ByteBuffer> mergeBufferPool;
 
   @Inject
   public QueryCountStatsMonitor(
-      QueryCountStatsProvider statsProvider
+      QueryCountStatsProvider statsProvider,
+      @Merging BlockingPool<ByteBuffer> mergeBufferPoolIn
```

Comment:
Why is it named `mergeBufferPoolIn`?

Reply:
to follow other code patterns, changed to 

```
  @Inject
  public QueryCountStatsMonitor(
      QueryCountStatsProvider statsProvider,
      @Merging BlockingPool<ByteBuffer> mergeBufferPool
  )
  {
    this.statsProvider = statsProvider;
    this.mergeBufferPool = mergeBufferPool;
  }

```